### PR TITLE
feat(infra-pipie): Added Infrastructure pipeline

### PR DESCRIPTION
**Pull request by carlosgit2016**
- Pull request trigger
- Push trigger
- Keep one plan comment in the PR

### CODE REVIEW

File: .github/workflows/infrastructure.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+name: infrastructure
+run-name: ${{ github.author }}
+
+on:
+  push:
+    branches:
+      - "main"
+  pull_request:
+    # paths:
```
Comment (carlosgit2016) [inline .github/workflows/infrastructure.yml:9]: Remove this after testing

File: .github/workflows/infrastructure.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    with:
+      pr: true
+      fmt-check: true
+      workspace: dev
+      working-dir: infrastructure/
+      role-to-assume: ${{ vars.GH_IDP_AWS_ROLE_TO_ASSUME }}
+
+  apply:
+    uses: ./.github/workflows/terraform.yml
+    if: github.event_name != 'push' && github.ref_name == 'main'
```
Comment (carlosgit2016) [inline .github/workflows/infrastructure.yml:34]: Tested here https://github.com/conquer-project/conquer_platform/actions/runs/6595925137/job/17921447706#step:12:1

Comment (benhur-araujo) [inline .github/workflows/infrastructure.yml:34]:
```suggestion
    if: github.ref_name == 'main'
```

File: .github/workflows/terraform.yml
Comment (benhur-araujo) [inline .github/workflows/terraform.yml:1]: Niceee

**Review by benhur-araujo — CHANGES_REQUESTED** [review]

**Review by benhur-araujo — APPROVED** [review]

**Pull request merged**